client/components/app.jsx

```diff
@@ -46,12 +52,44 @@ class App extends React.Component {
     )
       .then(res => {
         this.setState({
-          grades: this.state.grades.filter(grade => grade.id !== gradeId)
+          grades: this.state.grades.filter(grade => grade.id !== gradeId),
+          gradeToEditId: null
         });
       })
       .catch(error => console.error(error));
   }
 
+  editGrade(editGrade) {
+    fetch(
+      `/api/grades/${editGrade.id}`,
+      {
+        method: 'PATCH',
+        headers: {
+          'Content-Type': 'application/json'
+        },
+        body: JSON.stringify(editGrade)
+      }
+    )
+      .then(res => res.json())
+      .then(resGrade => this.setState({
+        grades: this.state.grades.map(stateGrade => {
+          return stateGrade.id !== resGrade.id
+            ? stateGrade
+            : resGrade;
+        }),
+        gradeToEditId: null
+      }))
+      .catch(error => console.error(error));
+  }
+
+  gradeToEdit(gradeId) {
+    this.setState({
+      grades: this.state.grades,
+      gradeToEditId: gradeId
+    });
+    return this.state.grades.filter(grade => grade.id === gradeId)[0];
+  }
```

Comment:
This method violates a principle known as "Command / Query Separation". A method should either be a command or a query, but not both. Basically, in most situations, a method should either perform a side-effect (such as updating the state of the system) _or_ answer a question, but never both. This method is really two methods that should be called in different places.

`setGradeToEdit` and `getGradeToEdit`

`setGradeToEdit` should be called as a result of clicking an update button
`getGradeToEdit` should be called as the app renders the form. I recommend returning `null` from `getGradeToEdit` if there is no `gradeToEditId` in state.

Straightening this out may help you identify and/or eliminate the cause of your bug.